miner/src/worker/mod.rs

```diff
@@ -41,6 +43,23 @@ impl WorkerController {
     }
 }
 
+fn partition_nonce(id: u64, total: u64) -> Range<u64> {
```

Comment:
integer division is round down, we can skip overflow checking:

```suggestion
fn partition_nonce(id: u64, total: u64) -> Range<u64> {
    debug_assert!(id < total);
    let span = u64::max_value() / total;
    let start = span * id;
    let end = start + span;
    (start..end)
}
```

Reply:
There still are few nonces in `end..u64::max_value()`.

Maybe the lucky nonce is one of them.

Reply:
Get, thanks.

btw, why set the parameter `id` and `total` type to u64? doesn't it should be usize?

Reply:
The output `nonce range` should be `u64`.

I think, both `u64` and `usize` are OK.
If I set `id` and `total` to `usize`, I can do cast at begin of the function `partition_nonce(..)`.